src/components/transactions/SortModal.tsx

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import Modal from 'react-native-modal';
+import { TouchableOpacity } from 'react-native';
+import {
+  CloseButtonContainer,
+  SortModalContainer,
+  Styles,
+  SortVerticalSpacing,
+  CenteredContainer,
+  PaddedScrollView,
+} from './styles';
+import {
+  BlueText,
+  Body1Text,
+  ButtonTextWhite,
+  H4CardNavTab,
+} from '../../../assets/Fonts';
+import RadioButton from '../common/RadioButton';
+import {
+  SortTransactionDispatch,
+  SortTransactionState,
+  SortVoucherDispatch,
+  SortVoucherState,
+} from '../../utils/transactionUtils';
+import { ButtonMagenta } from '../../../assets/Components';
+
+type SortModalProps = {
+  name: 'invoices' | 'vouchers';
+  isVisible: boolean;
+  setIsVisible: (visibility: boolean) => void;
+  sortState: SortTransactionState | SortVoucherState;
+  sortDispatch: SortTransactionDispatch | SortVoucherDispatch;
+  sortDescriptions: string[];
+};
+export default function SortModal({
+  name,
+  isVisible,
+  setIsVisible,
+  sortState,
+  sortDispatch,
+  sortDescriptions,
+}: SortModalProps) {
+  return (
+    <Modal
+      isVisible={isVisible}
+      coverScreen={false}
+      style={Styles.modal}
+      backdropTransitionOutTiming={0}
+    >
+      <SortModalContainer>
+        <CloseButtonContainer>
+          <TouchableOpacity
+            onPress={() => {
+              setIsVisible(false);
+              sortDispatch({ type: 'RESET_IN_PROGRESS' });
+            }}
+          >
+            <BlueText>
+              <Body1Text>Close</Body1Text>
+            </BlueText>
+          </TouchableOpacity>
+        </CloseButtonContainer>
```

Comment:
When you open the sort modal and apply changes without selecting a sort option, it claims: `Sort by: undefined` within the button. It then appears to sort invoice amounts from high to low.